Pull in upstream updates from cisagov/skeleton-ansible-role

jsf9k opened the pull request:
## 🗣 Description

In this pull request I merge in the develop branch from [cisagov/skeleton-ansible-role](https://github.com/cisagov/skeleton-ansible-role).  (This repository is so old it predates the skeletons!)

I also remove some code that was a workaround for ansible/ansible#56583, since that issue has been resolved and the workaround is no longer needed.

Finally, I make the code behave correctly and pass tests fro Fedora 29 and 31.

## 💭 Motivation and Context

I wanted to remove the workaround, so I may as well bring this repo into the modern world as well.

## 🧪 Testing

All pre-commit hooks and GitHub checks pass.

## 🚥 Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (causes existing functionality to change)

## ✅ Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

jsf9k committed:
First commit

jsf9k committed:
Remove molecule test from Travis CI config

jsf9k committed:
Rename repo to skeleton-generic

jsf9k committed:
Merge pull request #1 from cisagov/improvement/rename

Rename repo to skeleton-generic

jsf9k committed:
Remove author information from README

jsf9k committed:
Merge pull request #2 from cisagov/improvement/remove_author_from_readme

Remove author information from README

jsf9k committed:
Add/modify/remove files, make the skeleton specific to Ansible roles

jsf9k committed:
Fix typo in playbook.yml

jsf9k committed:
Add a comment for the dummy test, so folks know why it's dumb

jsf9k committed:
Merge pull request #1 from cisagov/improvement/add_comment_for_dummy_test

Add a comment for the dummy test, so folks know why it's dumb

jsf9k committed:
Add LGTM build badges

jsf9k committed:
Merge pull request #2 from cisagov/improvement/add_lgtm_build_badges

Add LGTM build badges

jsf9k committed:
Update the pyenv installation instructions

In particular, mention pyenv-installer.

jsf9k committed:
Merge pull request #3 from cisagov/improvement/update_pyenv_instructions

Update the pyenv installation instructions

jsf9k committed:
Merge remote-tracking branch 'skeleton-generic/develop' into improvement/merge_latest_changes_from_skeleton_generic

jsf9k committed:
Merge pull request #3 from cisagov/improvement/merge_latest_changes_from_skeleton_generic

Merge latest changes from skeleton-generic

jsf9k committed:
Add pre-commit hook for docker-compose linting

jsf9k committed:
Remove black and pyupgrade from requirements-test

They are not needed there.

Also run pre-commit autoupdate to pick up the latest version of hooks.

jsf9k committed:
Merge pull request #4 from cisagov/improvement/add_precommit_hook_for_docker_compose

Add pre-commit hook for docker-compose linting

jsf9k committed:
Add latest changes from skeleton-generic

jsf9k committed:
Merge pull request #4 from cisagov/improvement/add_latest_changes_from_generic_skeleton

Add latest changes from skeleton-generic

jsf9k committed:
Add bandit pre-commit hook

Also ran pre-commit autoupdate, which picked up a newer version of the
pyupgrade hook.

jsf9k committed:
Merge pull request #5 from cisagov/improvement/add_bandit

Add bandit pre-commit hook

jsf9k committed:
Merge remote-tracking branch 'skeleton-generic/develop' into improvement/merge_in_skeleton_generic_changes

jsf9k committed:
Turn off bandit for the molecule test code

Bandit complains about the use of assert() in test code.

jsf9k committed:
Merge pull request #5 from cisagov/improvement/merge_in_skeleton_generic_changes

Merge in skeleton-generic changes

jsf9k committed:
Ignore W503 from flake8

This is a warning about line breaks before binary operators.  It no
longer agrees with PEP8.  See, for example, here:
https://github.com/ambv/black/issues/21.

Guido agrees here:
https://github.com/python/peps/commit/c59c4376ad233a62ca4b3a6060c81368bd21e85b.

jsf9k committed:
Merge pull request #6 from cisagov/improvement/disable_w503

Ignore W503 from flake8

jsf9k committed:
Merge remote-tracking branch 'skeleton-generic/develop' into improvement/merge_latest_skeleton_generic_changes

jsf9k committed:
Merge pull request #6 from cisagov/improvement/merge_latest_skeleton_generic_changes

Improvement/merge latest skeleton generic changes

jsf9k committed:
Add pydoc warnings to flake8 config

Also add flake8-docstrings as an additional dependency in the
.pre-commit-config.yml file.

jsf9k committed:
Merge pull request #7 from cisagov/improvement/enable_docstring

Add pydoc warnings to flake8 config

jsf9k committed:
Merge remote-tracking branch 'skeleton-generic/develop' into improvement/merge_in_upstream_changes_from_skeleton_generic

jsf9k committed:
Make the new pydoc flake8 rules happy

jsf9k committed:
Merge pull request #7 from cisagov/improvement/merge_in_upstream_changes_from_skeleton_generic

Merge in upstream changes from skeleton-generic

felddy committed:
Add a configuration file for Bandit, and configure pre-commit to use it.

felddy committed:
Improve english

felddy committed:
Merge pull request #8 from cisagov/improvement-bandit-config

Improvement bandit config

jsf9k committed:
Run pre-commit autoupdate

Also update the URL for black since it moved GitHub orgs from ambv to
python.

jsf9k committed:
Merge branch 'develop' into improvement-bandit-config

jsf9k committed:
Merge pull request #9 from cisagov/improvement-bandit-config

Run pre-commit autoupdate

jsf9k committed:
Merge remote-tracking branch 'skeleton-generic/develop' into improvement/pull_in_upstream_changes_from_skeleton_generic

jsf9k committed:
Make .bandit.yml pass yamllint

jsf9k committed:
Merge pull request #10 from cisagov/improvement/make_bandit_config_pass_yamllint

Make .bandit.yml pass yamllint

jsf9k committed:
Merge remote-tracking branch 'skeleton-generic/develop' into improvement/pull_in_upstream_changes_from_skeleton_generic

jsf9k committed:
Merge pull request #8 from cisagov/improvement/pull_in_upstream_changes_from_skeleton_generic

Pull in upstream changes from skeleton-generic

jsf9k committed:
Modify requirements file to use the latest molecule from master.

This is a temporary measure, taken because the latest release of
molecule does not play well with ansible 2.8, which was just
released.  (Molecule installs the latest version of Ansible as a
dependency.)

We will revert this change once a new release of molecule comes out.

jsf9k committed:
Add missing flake8 dependency

The bleeding-edge version of molecule is apparently missing the flake8
dependency.

jsf9k committed:
Merge pull request #9 from cisagov/bugfix/use_bleeding_edge_molecule

Modify requirements file to use the latest molecule from master.

jsf9k committed:
Enable TravisCI caching for pip and pre-commit plugins

This should speed up TravisCI builds.

jsf9k committed:
Update pre-commit plugins to latest versions

Via pre-commit autoupdate.

jsf9k committed:
Merge pull request #11 from cisagov/improvement/use_travisci_caching

Use TravisCI caching to speed up builds

jsf9k committed:
Merge remote-tracking branch 'skeleton-generic/develop' into improvement/upstream_updates

jsf9k committed:
Merge pull request #10 from cisagov/improvement/upstream_updates

Upstream updates

jsf9k committed:
Add changes from running pre-commit autoupdate

jsf9k committed:
Merge pull request #12 from cisagov/improvement/update_precommit

Add changes from running pre-commit autoupdate

jsf9k committed:
Remove the terraform-docs hook

terraform-docs is currently broken with respect to terraform 0.12.

jsf9k committed:
Merge pull request #13 from cisagov/bugfix/turn_off_terraform_docs_hook

Remove the terraform-docs hook

jsf9k committed:
Merge remote-tracking branch 'skeleton-generic/develop' into improvement/merge_in_upstream_changes

jsf9k committed:
Merge pull request #11 from cisagov/improvement/merge_in_upstream_changes

Merge in upstream changes from cisagov/skeleton-generic

jsf9k committed:
Configure prettier to ignore JSON files

JSON files are already being linted by the pretty-format-json hook.

jsf9k committed:
Merge pull request #14 from cisagov/improvement/tell_prettier_to_ignore_json_files

Configure prettier to ignore JSON files

jsf9k committed:
Merge remote-tracking branch 'skeleton-generic/develop' into improvement/pull_in_upstream_changes

jsf9k committed:
Merge pull request #12 from cisagov/improvement/pull_in_upstream_changes

Pull in upstream changes

jsf9k committed:
Add the docker extra to the installation of molecule

This gets rid of the need to install the docker python module
separately.

flake8 is now a dependency of the latest molecule, so there is no need
to install it separately either.

jsf9k committed:
Merge pull request #13 from cisagov/improvement/better_installation_of_molecule

Add the docker extra to the installation of molecule

mcdonnnj committed:
Update two text blocks to use the console tag instead of the bash tag for formatting.

mcdonnnj committed:
Merge pull request #15 from cisagov/update_contributing_text_blocks

Update two text blocks to use the console tag instead of the bash tag…

mcdonnnj committed:
Merge upstream changes in skeleton-generic.

mcdonnnj committed:
Merge pull request #14 from cisagov/merge_skeleton-generic_updates

Merge skeleton generic updates

jsf9k committed:
Update pre-commit hooks via pre-commit autoupdate

jsf9k committed:
Add config file for markdownlint

This config file turns off the "line too long" rule when within a code
block or a table.  Often there is no way to avoid going over 80 lines
in such cases.

jsf9k committed:
Rename LICENSE.md to LICENSE, make the other changes that requires

The license text is not actually in Markdown, and adding the .md
extension actually makes it render a little differently than intended.

jsf9k committed:
Merge pull request #16 from cisagov/improvement/markdownlint_config

Add markdownlint config

jsf9k committed:
Merge remote-tracking branch 'skeleton-generic/develop' into improvement/upstream_changes_from_skeleton_generic

jsf9k committed:
Merge pull request #15 from cisagov/improvement/upstream_changes_from_skeleton_generic

Merge in upstream changes from cisagov/skeleton-generic

jsf9k committed:
Add support for Debian Buster

Buster is scheduled to be released today.  Buster also includes the
freeipa-client package, while Stretch does not, so switching to Buster
would make our lives a lot easier going forward.

jsf9k committed:
Merge pull request #16 from cisagov/improvement/add_buster

Add support for Debian Buster

jsf9k committed:
Use a prerelease of molecule instead of the latest from master

The latest from master is currently broken, but the latest prerelease
has all the goodies we need.  Once a new release is made we will use
that instead.

jsf9k committed:
Merge pull request #17 from cisagov/bugfix/upstream_molecule_broken

Use a prerelease of molecule instead of the latest from master

jsf9k committed:
Add updated Dockerfile.j2 and INSTALL.rst files

These skeleton files come from the latest release of molecule,
generated via the command "molecule init scenario".

jsf9k committed:
Add latest .yamllint generated via molecule init scenario

The generated file contains the truthy: disable bit that was in our
old file.

jsf9k committed:
Add fedora 29 and 30 to molecule configuration

jsf9k committed:
Trigger LGTM

jsf9k committed:
Merge pull request #18 from cisagov/improvement/updated_skeleton_files_from_latest_molecule

Add updated "molecule init scenario" files

jsf9k committed:
Add updates from running pre-commit autoupdate

jsf9k committed:
Merge pull request #17 from cisagov/improvement/updates_from_precommit_autoupdate

Add updates from running pre-commit autoupdate

jsf9k committed:
Merge remote-tracking branch 'skeleton-generic/develop' into improvement/upstream_updates

jsf9k committed:
Merge pull request #19 from cisagov/improvement/upstream_updates

Upstream updates

jsf9k committed:
Change "company" name to the latest trend

The new name agrees with the naming guidance that was handed down by
Bobby Thompson.

jsf9k committed:
Merge pull request #20 from cisagov/change_company_name

Change "company" name to the latest trend

dav3r committed:
Add updates from running pre-commit autoupdate

dav3r committed:
Merge pull request #18 from cisagov/improvement/updates_from_precommit_autoupdate

Add updates from running pre-commit autoupdate

dav3r committed:
Merge remote-tracking branch 'skeleton-generic/develop' into improvement/merge_in_upstream_changes

felddy committed:
Replace Travis-CI with GitHub actions.

felddy committed:
Fix action shield link

felddy committed:
Disable ansible-lint until they fix their setup.py
See: https://github.com/ansible/ansible-lint/issues/590

dav3r committed:
Merge pull request #21 from cisagov/improvement/merge_in_upstream_changes

Merge in upstream changes

felddy committed:
Remove extra line.

felddy committed:
Merge pull request #20 from cisagov/improvement/actions

Replace Travis-CI with GitHub actions

felddy committed:
Merge branch 'develop' of github.com:cisagov/skeleton-generic into improvement/actions

felddy committed:
Add molecule run

felddy committed:
Remove white space that makes some of the team itchy.

felddy committed:
Merge pull request #22 from cisagov/improvement/actions

Replace Travis-CI with GitHub actions

mcdonnnj committed:
Re-enabled ansible-lint and ran pre-commit autoupgrade.

mcdonnnj committed:
Merge pull request #21 from cisagov/pre-commit_autoupdate_2019-11-02

Re-enable ansible-lint and autoupgrade.

felddy committed:
Merge branch 'develop' of github.com:cisagov/skeleton-generic into improvement/upstream

felddy committed:
Remove vestigial travis file.

felddy committed:
Add GitHub action caching of pre-commit hooks and pip packages.

felddy committed:
Merge branch 'develop' into improvement/action_cache

felddy committed:
Remove search for previous pre-commit caches.
These seem to be very large caches.  Restoring an old one and updating 
it results in a cache larger than the maximum allowed cache size: 200MB. 
 
"Cache size of 254757924 bytes is over the 200MB limit, not saving 
cache."
So if the config changes it is best to just take the cache-miss and 
start from scratch.

felddy committed:
Merge pull request #23 from cisagov/improvement/upstream

Pull in upstream: pre-commit updates, and clean out travis

felddy committed:
Merge pull request #22 from cisagov/improvement/action_cache

Add GitHub action caching of pre-commit hooks and pip packages.

felddy committed:
Add GitHub Action caches for pip and pre-commit.

felddy committed:
Split build workflow into lint and test jobs.
This should speed things up a bit since pre-commit checks are slow, and 
so is the molecule run.

felddy committed:
Merge pull request #25 from cisagov/improvement/action_caches

Add GitHub Action caches for pip and pre-commit, split workflow into two jobs.

felddy committed:
Force pre-commit hooks to use python3. 
Fixes #23

felddy committed:
Autoupdate pre-commit hooks.

felddy committed:
Merge pull request #24 from cisagov/improvement/pre_commit_default_python

Set a default language for pre-commit hooks

felddy committed:
Merge branch 'develop' of github.com:cisagov/skeleton-generic into improvement/upstream

felddy committed:
Merge pull request #26 from cisagov/improvement/upstream

Improvement/upstream

felddy committed:
Add codeowners file with team OIS maintainers.
See: 
https://help.github.com/en/github/creating-cloning-and-archiving-repositories/about-code-owners

felddy committed:
Merge pull request #25 from cisagov/improvement/codeowners

Add codeowners file with team OIS maintainers.

felddy committed:
Update CODEOWNERS

felddy committed:
Merge branch 'develop' of github.com:cisagov/skeleton-generic into improvement/codeowners

felddy committed:
Merge pull request #27 from cisagov/improvement/codeowners

Improvement/codeowners

mcdonnnj committed:
Add seed-isort-config and isort to pre-commit hooks.
Add isort configuration file.

mcdonnnj committed:
Merge pull request #28 from cisagov/improvements/pre-commit_add_isort

Add isort tool to pre-commit hooks.

jsf9k committed:
Make workflow run when a PR is opened, synchronized, or reopened

A user forked cisagov/scan-target-data and created a pull request, but
the required GitHub Action(s) did not run.  This is presumably because
the user does not have Actions enabled in his or her fork.  Ideally,
the required Action(s) would run in cisagov/scan-target-data when a PR
to merge changes back is created.  Based on my reading of this link,
adding the "pull_request" event type should make this happen:
https://help.github.com/en/actions/automating-your-workflow-with-github-actions/events-that-trigger-workflows#pull-request-events-for-forked-repositories

jsf9k committed:
Improve list formatting

jsf9k committed:
Merge pull request #29 from cisagov/improvement/better-handle-prs-from-forked-repos

Make workflow run when a PR is opened, synchronized, or reopened

jsf9k committed:
Merge remote-tracking branch 'skeleton-generic/develop' into improvement/upstream-changes

jsf9k committed:
Add .isort.cfg modified by pre-commit hook

jsf9k committed:
Commit changes made by isort pre-commit hook

jsf9k committed:
Merge pull request #28 from cisagov/improvement/upstream-changes

Pull in upstream changes

mcdonnnj committed:
Backported changes to CONTRIBUTING.md from the development guide.

mcdonnnj committed:
Merge pull request #31 from cisagov/improvements/add_wsl_instructions

Cleanup Linux instructions and add WSL references to CONTRIBUTING.md

mcdonnnj committed:
Update Python version used to 3.8
Update actions/checkout to v2
Update formatting to match downstream children

mcdonnnj committed:
Merge pull request #32 from cisagov/improvements/update_python_version_and_checkout_action

Update Python and actions/checkout Versions

mcdonnnj committed:
Run pre-commit autoupdate.

mcdonnnj committed:
Flip cache order to mirror how it is done downstream.

mcdonnnj committed:
Merge pull request #33 from cisagov/improvement/update_pre-commit_hooks

Update pre-commit hooks (and flip Actions cache order in workflow).

mcdonnnj committed:
Merge branch 'develop' of github.com:cisagov/skeleton-generic into improvements/pull_in_upstream_changes

mcdonnnj committed:
Update molecule version to the recent preleases for v3.0 as they support recent versions of Ansible. This has also required updating the molecule.yml file to align with 3.0 changes.

mcdonnnj committed:
All references to '-r' for pip calls have been replaced with the more verbose '--requirement'.

mcdonnnj committed:
Adjust spacing for lint directive parameters.

mcdonnnj committed:
Add pip list to see what packages are installed.

mcdonnnj committed:
Pin molecule to <3.
Revert changes to molecule.yml

mcdonnnj committed:
Merge pull request #34 from cisagov/improvements/use_verbose_pip_switches

Use Verbose pip Switches

mcdonnnj committed:
Remove pip list debugging.

mcdonnnj committed:
Merge branch 'develop' of github.com:cisagov/skeleton-generic into improvements/pull_in_upstream_changes

mcdonnnj committed:
Update comment for molecule version pinning.

mcdonnnj committed:
Merge pull request #29 from cisagov/improvements/pull_in_upstream_changes

Pull in Upstream Changes

felddy committed:
Autoupdate pre-commit hooks.  Add mypy.

felddy committed:
Merge pull request #35 from cisagov/improvement/static_type_checking

Autoupdate pre-commit hooks.  Add mypy.

dav3r committed:
Add Fedora 31
Comment out Fedora 30 and include explanation why

dav3r committed:
Add Fedora 29 and 31.
Add comment explaining why Fedora 30 cannot be used.

dav3r committed:
Sort platforms alphabetically

dav3r committed:
Merge pull request #30 from cisagov/improvement/support_fedora_31

Support Fedora 31

jsf9k committed:
Fix some indenting

This was driving me nuts!

jsf9k committed:
Merge pull request #31 from cisagov/improvement/fix-indenting

Fix some indenting

jsf9k committed:
Merge remote-tracking branch 'skeleton-generic/develop' into improvement/upstream-skeleton-updates

jsf9k committed:
Add .mypy_cache to .gitignore

jsf9k committed:
Add .mypy_cache to .gitignore

jsf9k committed:
Merge pull request #36 from cisagov/improvement/add-mypy-cache-to-gitignore

Add .mypy_cache to .gitignore

jsf9k committed:
Merge pull request #32 from cisagov/improvement/upstream-skeleton-updates

Pull in upstream skeleton updates from skeleton-generic

jsf9k committed:
Merge remote-tracking branch 'skeleton-ansible-role/develop' into improvement/upstream-skeleton-updates

jsf9k committed:
Remove workaround for ansible/ansible#56583

That issue has been resolved.

jsf9k committed:
Refine code that modifies /etc/logrotate.d/rsyslog

Also make the test code pass for fedora.  Previously it was not.

felddy reviewed (approved):
This is a time capsule!  ⏳ 💊

dav3r commented on tasks/main.yml:
```diff
@@ -38,19 +33,35 @@
     # Keep seven days worth of logs
-    - regex: '\s*rotate'
-      line: '\trotate 7'
+    - regex: '\s*rotate \d+$'
```
Is there a reason this `regex` and the one above it were not switched from single quotes to double quotes or was this just an oversight?

jsf9k replied:
What is there is correct.  This is YAML foo.

See [here](https://docs.ansible.com/ansible/latest/reference_appendices/YAMLSyntax.html), @dav3r.  I needed the double quotes for the strings containing escapes (`\t`).  The regexes don't need double quotes since they don't contain escapes; `\s` is a regex ~thing~thang, not an escape.

Let me know if this satisfies your question.  I will wait to merge until I hear back from you.

dav3r replied:
Yes, that makes sense, but it looks odd because we have a seemingly-inconsistent mix of single-quoting and double-quoting throughout this file (more than just these two regex lines).

jsf9k replied:
Well it looks odd, but thus is YAML.  I hesitate to add comments to these lines since I think a decent understanding of YAML's warts should be assumed when working with Ansible.

I will add a comment if you can convince @felddy or @mcdonnnj, though.  I will defer to the wisdom of the herd.

dav3r replied:
I agree with you and don't think that a comment is needed here either.  Although if we are only double-quoting when necessary, then shouldn't these lines be using single quotes?
- https://github.com/cisagov/ansible-role-logrotate/blob/improvement/upstream-skeleton-updates/tasks/main.yml#L53
- https://github.com/cisagov/ansible-role-logrotate/blob/improvement/upstream-skeleton-updates/tasks/main.yml#L63
- https://github.com/cisagov/ansible-role-logrotate/blob/improvement/upstream-skeleton-updates/tasks/main.yml#L65
- https://github.com/cisagov/ansible-role-logrotate/blob/improvement/upstream-skeleton-updates/tasks/main.yml#L67

mcdonnnj replied:
I agree with @dav3r about consistency. I think double quotes and escaping the backslash (`"\\s*rotate \\d+$"`) is more productive than a comment explaining single vs double quotes if we went that route however.

jsf9k replied:
@dav3r and @mcdonnnj, try on commit 30725fe for size.  Let me know what you think.

dav3r replied:
That works for me- thanks and sorry for the extra work!

jsf9k replied:
Once @mcdonnnj is happy I will merge.

jsf9k replied:
@mcdonnnj confirmed to me that he is happy with the quotes.

dav3r reviewed (approved):
👍 though see my one question.

jsf9k committed:
Make quotes consistent

I changed this file to use double quotes everywhere, except for a few
regexes that contain substrings like "\s".  For those regexes I also
added a comment to explain why I strayed from double quotes.

jsf9k committed:
Merge remote-tracking branch 'skeleton-generic/develop' into improvement/switch-to-amazonlinux-2

jsf9k committed:
Switch to Amazon Linux 2

Amazon Linux 1 is ancient stuff at this point.

jsf9k committed:
Merge pull request #33 from cisagov/improvement/switch-to-amazonlinux-2

Switch to amazonlinux 2

jsf9k committed:
Merge remote-tracking branch 'skeleton-ansible-role/develop' into improvement/upstream-skeleton-updates
